Source/OxyPlot/Series/XYAxisSeries.cs

```diff
@@ -750,62 +750,65 @@ protected int UpdateWindowStartIndex<T>(IList<T> items, Func<T, double> xgetter,
         /// <param name="targetX">target x.</param>
         /// <param name="initialGuess">initial guess index.</param>
         /// <returns>
-        /// index of x with max(x) &lt;= target x or -1 if cannot find
+        /// index of x with max(x) &lt;= target x or 0 if cannot find
         /// </returns>
-        protected int FindWindowStartIndex<T>(IList<T> items, Func<T, double> xgetter, double targetX, int initialGuess)
+        public int FindWindowStartIndex<T>(IList<T> items, Func<T, double> xgetter, double targetX, int initialGuess)
         {
-            int lastguess = 0;
             int start = 0;
-            int end = items.Count - 1;
-            int curGuess = initialGuess;
+            int nominalEnd = items.Count - 1;
+            while (nominalEnd > 0 && double.IsNaN(xgetter(items[nominalEnd])))
+                nominalEnd -= 1;
+            int end = nominalEnd;
+            int curGuess = Math.Max(0, Math.Min(end, initialGuess));
 
-            while (start <= end)
+            double GetX(int index)
```

Comment:
Presently the method can return indexes to `NaN` values, which is inconsistent with the documentation, because this pretends the item at the current index is something that it is not. Returning the 'true' index might be a sufficient solution; however, this value could not be used for updating the `end`, only the start or returning the index when found.

Reply:
I thought about that. 

If the use case is that we want to know which is the first index we need to consider in order to draw the graph correctly, we could make the function return the first index that needs to be drawn. The could well be an index with X-value larger than the targetX.

An alternative is to define the function to return the largest non-NaN value with x < targetX.

I think I prefer the latter alternative. What do you think?

Reply:
Yeah, that latter idea is how I would read the documentation (i.e. find the index with largest x value that is no more than the target, which arguably can't be a NaN entry).